README.md

```diff
@@ -63,1498 +104,468 @@ Delegating a capability is like the statement `message`. Task is akin to `messag
+ 
+A Task MUST be uniquely defined by a Task ID that is the CID of the following fields:
 
-**NB: a Receipt does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
+- [Subject]
+- [Command]
+- [Arguments]
+- [Nonce]
 
-Receipts MUST use the same version as the invocation that they contain.
+Tasks that describe pure functions — or other strategies like fan-out racing — SHOULD have the same Task ID by using the same nonce.
 
-## 8.1 Schema
+#### Command
+[Command]: #command
 
-```ipldsch
-type Receipt struct {
-  ran     &Invocation
+The REQUIRED Command (`cmd`) field MUST contain a concrete, dispatchable message that can be sent to the Executor. The Command MUST define the shape of the data in the [Arguments].
 
-  # output of the invocation
-  out     Result
+### Arguments
+[Arguments]: #arguments
 
-  # Effects to be performed
-  fx      Effects
+The REQUIRED Arguments (`args`) field, MAY contain any parameters expected by the Command. The Subject MUST be considered the authority on the shape of this data. This field MUST be representable as a map or keyword list.
 
-  # All the other metadata
-  meta    {String: any}
+The Arguments MUST pass validation of the Policies on all of the [UCAN Delegations][UCAN Delegation] in the [Proofs] field. If any Policy reports failure, the Invocation MUST be rejected.
 
-  # Principal that issued this receipt.
-  # If omitted issuer is inferred from 
-  # the invocation task audience.
-  iss     optional Principal
+#### Nonce
+[Nonce]: #nonce
 
-  # When issuer is different from executor
-  # this MUST hold a UCAN delegation chain 
-  # from executor to the issuer.
-  # This field SHOULD be omitted executor is an issuer.
-  prf     [&UCAN]
+The REQUIRED `nonce` field MUST include a random nonce. This field ensures that multiple (non-idempotent) invocations are unique. The nonce SHOULD be empty (`0x`) for Commands that are idempotent (such as deterministic Wasm modules or standards-abiding HTTP PUT requests).
 
-  # Signature from the `iss`.
-  s       Varsig
-}
-```
+### Proofs
+[Proofs]: #proofs
 
-## 8.2 Fields
+The `prf` field defines all [Delegation]s required to prove that this Invocation has an unbroken authorization chain.
 
-### 8.2.1 Ran Invocation
+### Provenance
+[Provenance]: #provenance
 
-The `ran` field MUST include a link to the [Invocation] that the Receipt is for.
+#### Cause
+[Cause]: #cause
 
-### 8.2.2 Output
+The OPTIONAL `cause` field is a provenance claim describing which [Receipt] requested it. This is helpful for tracking chains of Invocations.
 
-The `out` field MUST contain the value output of the invocation in [Result] format.
+#### Expiration
+[Expiration]: #expiration
 
-### 8.2.3 Effect
+The OPTIONAL field `exp` defines when the Invocation SHOULD time out. This is both expressive (defines a timeout, which is a best practice), and prevents replays.
```

Comment:
But `exp` is still [REQUIRED in the high-level UCAN spec](https://github.com/ucan-wg/spec/blob/9b50374ba600eea56a697f204f870cd36fba37c7/README.md#payload)?

Reply:
Fixed